### PR TITLE
REM-407

### PR DESCRIPTION
**Pull request by imtithalhalalo**

### CODE REVIEW

File: src/pages/pu-settings/Form/PUSettingsForm.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  const { labels: _labels, access } = useResourceQuery({
+    datasetId: ResourceIds.PUSettings
+  })
+
+  const arrayAllow = [
+    'PUCurrencyId',
+    'PUSiteId',
+    'PUFIIntegration',
+    'POSHPVarPct'
+  ]
```
Comment (Alinadimmokdad) [inline src/pages/pu-settings/Form/PUSettingsForm.js:31]: you can directly put them in submit , no need for a new array

File: src/pages/pu-settings/Form/PUSettingsForm.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { useForm } from 'src/hooks/form'
+import FormShell from 'src/components/Shared/FormShell'
+import { ControlContext } from 'src/providers/ControlContext'
+import toast from 'react-hot-toast'
+
+const PUSettingsForm = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
```
Comment (omarZrayka23) [inline src/pages/pu-settings/Form/PUSettingsForm.js:20]: remove getRequest

Comment (imtithalhalalo) [reply src/pages/pu-settings/Form/PUSettingsForm.js:20]: it is needed for getting the data in the first load

File: src/pages/pu-settings/Form/PUSettingsForm.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        })
+
+        if (response) {
+          toast.success(platformLabels.Added)
+        }
+      } catch (error) {}
+    }
+  })
+
+  const getData = async () => {
```
Comment (omarZrayka23) [inline src/pages/pu-settings/Form/PUSettingsForm.js:71]: useEffect(() => {
    ;(async function () {
  })()
  }, [])

File: src/pages/pu-settings/Form/PUSettingsForm.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        }
+      })
+
+    } catch (error) {}
+  }
+  
+  useEffect(() => {
+    getData()
+  }, [access])
+
```
Comment (Alinadimmokdad) [inline src/pages/pu-settings/Form/PUSettingsForm.js:91]: do we need [access]?

Comment (imtithalhalalo) [reply src/pages/pu-settings/Form/PUSettingsForm.js:88]: no removed

File: src/pages/pu-settings/Form/PUSettingsForm.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  })
+  
+  useEffect(() => {
+    ;(async function () {
+      try {
+        const response = await getRequest({
+          extension: SystemRepository.Defaults.qry,
+          parameters: `_filter=`
+        })
+        const myObject = {}
```
Comment (omarZrayka23) [inline src/pages/pu-settings/Form/PUSettingsForm.js:78]: are u need?

File: src/pages/pu-settings/Form/PUSettingsForm.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        response.list.forEach(obj => {
+          if (arrayAllow.includes(obj.key)) {
+            myObject[obj.key] = obj.key ? parseInt(obj.value) : null
```
Comment (omarZrayka23) [inline src/pages/pu-settings/Form/PUSettingsForm.js:81]: why used it? if u do not need clear

**Review by omarZrayka23 — CHANGES_REQUESTED** [review]

**Pull request merged**